syntax = "proto3";

package ocr;

service OcrService {
    rpc RecognizeImage (OcrRequest) returns (OcrResponse);
}

message OcrRequest {
    int64 batch_id = 1;
    int32 image_index = 2;
    string filename = 3;
    bytes image_data = 4;
}

message OcrResponse {
    int64 batch_id = 1;
    int32 image_index = 2;
    string filename = 3;
    string text = 4;
    bool success = 5;
    string error_message = 6;
    int64 processing_time_ms = 7;
}
